source/server/server.cc

```diff
@@ -388,7 +389,7 @@ RunHelper::RunHelper(Event::Dispatcher& dispatcher, Upstream::ClusterManager& cm
   // this can fire immediately if all clusters have already initialized. Also note that we need
   // to guard against shutdown at two different levels since SIGTERM can come in once the run loop
   // starts.
-  cm.setInitializedCb([this, &init_manager, workers_start_cb]() {
```

Comment:
I think the scope here is too wide. Won't this prevent listeners warming? I think ideally we pause/resume around just this loop: https://github.com/envoyproxy/envoy/blob/c223232bbff5aadde8ec83a548e18f99d4729914/source/server/configuration_impl.cc#L54

Reply:
I tried that at first but it didn't seem to work - while the listeners are created there the RDS subscription isn't started until the initialize callback. Pausing just around the static listener loop mean that we resumed before the subscriptions started.

It won't block the listeners from warming because I'm resuming after all `initialize` callbacks are called, not when initialization has been complete. Doing the resume inside the callback passed to `init_manager` does prevent the listeners from warming, presumably because some init callback is waiting for the routes. 

Happy to consider another approach - I'm not super happy about having the start and resume action be so far from each other.